src/Raven.Client/Document/SessionOperations/NewLoadOperation.cs

```diff
@@ -0,0 +1,147 @@
+using Raven.Client.Documents.Commands;
+using Sparrow.Json;
+using Sparrow.Logging;
+
+namespace Raven.Client.Document.SessionOperations
+{
+    public class NewLoadOperation
+    {
+        private readonly InMemoryDocumentSessionOperations _session;
+        private static readonly Logger _logger = LoggerSetup.Instance.GetLogger<NewLoadOperation>("Raven.Client");
+        private BlittableJsonReaderObject[] _result;
+
+        private string[] _ids;
+        private readonly List<string> _idsToCheckOnServer = new List<string>();
+
+        public NewLoadOperation(InMemoryDocumentSessionOperations session)
+        {
+            _session = session;
+        }
+
+        public GetDocumentCommand CreateRequest()
+        {
+            if (_idsToCheckOnServer.Count == 0)
+                return null;
+
+            _session.IncrementRequestCount();
+            if (_logger.IsInfoEnabled)
+                _logger.Info($"Requesting the following ids '{string.Join(", ", _idsToCheckOnServer)}' from {_session.StoreIdentifier}");
+            return new GetDocumentCommand
+            {
+                Ids = _idsToCheckOnServer.ToArray()
+            };
+        }
+
+        public void ById(string id)
+        {
+            if (id == null)
+                throw new ArgumentNullException(nameof(id), "The document id cannot be null");
+
+            _ids = new[] {id};
+            if (_session.IsLoadedOrDeleted(id))
+                return;
+
+            _idsToCheckOnServer.Add(id);
+        }
+
+        public void ByIds(IEnumerable<string> ids)
+        {
+            _ids = ids.ToArray();
+            foreach (var id in _ids
+                .Distinct(StringComparer.OrdinalIgnoreCase)
+                .ToArray())
+            {
+                ById(id);
+            }
+        }
+
+        public T GetDocument<T>()
+        {
+            return GetDocument<T>(_ids[0]);
+        }
+
+        private T GetDocument<T>(string id)
+        {
+            if (_session.IsDeleted(id))
+                return default(T);
+
+            object existingEntity;
+            if (_session.EntitiesById.TryGetValue(id, out existingEntity))
+                return (T) existingEntity;
+
+            BlittableJsonReaderObject document;
+            if (_session.DocumentsById.TryGetValue(id, out document) == false)
+                return default(T);
+
+            var entity = _session.ConvertToEntity(typeof(T), id, document);
+            try
+            {
+                _session.EntitiesById.Add(id, entity);
+            }
+            catch (Exception)
+            {
+                if (_logger.IsInfoEnabled)
+                    _logger.Info("Tried to add an exisitg entity");
+            }
+            return (T) entity;
+        }
+
+        public T[] GetDocuments<T>()
+        {
+            var finalResults = new T[_ids.Length];
+            for (int i = 0; i < _ids.Length; i++)
+            {
+                var id = _ids[i];
```

Comment:
I'm using the id to pass to GetDocument method. I'm not sure what is the issue you see here.